linera-views/src/reentrant_collection_view.rs

```diff
@@ -326,6 +326,168 @@ where
+                    let entry = entry.into_mut();
+                    if let Update::Removed = entry {
+                        selected_short_keys.push(short_key);
+                        selected_to_clear.push(true);
+                    }
+                }
+                btree_map::Entry::Vacant(_entry) => {
+                    selected_short_keys.push(short_key);
+                    selected_to_clear.push(self.was_cleared);
+                }
+            }
+        }
+        let mut handles = Vec::new();
+        for short_key in selected_short_keys.clone() {
+            let context = self.context.clone();
+            handles.push(tokio::spawn(async move {
+                let key = context.base_tag_index(KeyTag::Subview as u8, &short_key);
+                let context = context.clone_with_base_key(key);
+                W::load(context).await
+            }));
+        }
+        let response = futures::future::join_all(handles).await;
+        for (i, view) in response.into_iter().enumerate() {
+            let short_key = selected_short_keys[i].clone();
+            let mut view = view??;
+            if selected_to_clear[i] {
+                view.clear();
+            }
+            let wrapped_view = Arc::new(RwLock::new(view));
+            updates.insert(short_key.clone(), Update::Set(wrapped_view));
+        }
+        let mut result = Vec::new();
+        for short_key in short_keys {
+            result.push(
+                if let btree_map::Entry::Occupied(entry) = updates.entry(short_key) {
+                    let entry = entry.into_mut();
+                    if let Update::Set(view) = entry {
+                        view.clone().try_write_owned()?
+                    } else {
+                        unreachable!()
+                    }
+                } else {
+                    unreachable!()
+                },
+            );
+        }
+        Ok(result)
+    }
+
+    /// Load multiple entries for read at once.
+    /// The entries in short_keys have to be all distincts.
+    /// ```rust
+    /// # tokio_test::block_on(async {
+    /// # use linera_views::memory::{create_test_context, MemoryContext};
+    /// # use linera_views::reentrant_collection_view::ReentrantByteCollectionView;
+    /// # use linera_views::register_view::RegisterView;
+    /// # use crate::linera_views::views::View;
+    /// # let context = create_test_context();
+    ///   let mut view : ReentrantByteCollectionView<_, RegisterView<_,String>> = ReentrantByteCollectionView::load(context).await.unwrap();
+    ///   let short_keys = vec![vec![0, 1], vec![2, 3]];
+    ///   let subviews = view.try_load_entries(short_keys).await.unwrap();
+    ///   let value1 = subviews[0].get();
+    ///   let value2 = subviews[1].get();
+    ///   assert_eq!(*value1, String::default());
+    ///   assert_eq!(*value2, String::default());
+    /// # })
+    /// ```
+    pub async fn try_load_entries(
```

Comment:
Should define `try_load_entry` as a function of `try_load_entries`? (And same for `mut`) For one, this would improve the test coverage.

Reply:
I am against because the `try_load_entry` uses the `entry` in a more subtle way than the one for `load_entries`.

Reply:
Are you talking about a semantic difference or extra operations? Let's follow up on this

Reply:
There is no semantic difference. I just think that if you access with `try_load_entries` for one single entry then you access to the `BTreeMap` two times while for the `try_load_entry` you need just one time.
Of course it washes out compared to other slowdown.

Reply:
ok makes sense!